pkg/server/test/list_objects.go

```diff
@@ -51,6 +52,63 @@ func newListObjectsRequest(store, objectType, relation, user, modelID string, co
 func ListObjectsTest(t *testing.T, ds storage.OpenFGADatastore) {
 	ctx := context.Background()
 
+	t.Run("list_objects_with_concurrent_checks_and_contextual_tuples", func(t *testing.T) {
+		store := ulid.Make().String()
+
+		typedefs := parser.MustParse(`
+		type user
+
+		type doc
+		  relations
+		    define reader: [user] as self
+		    define blocked: [user] as self
+			define viewer as reader but not blocked
+		`)
+
+		model := &openfgapb.AuthorizationModel{
+			Id:              ulid.Make().String(),
+			SchemaVersion:   typesystem.SchemaVersion1_1,
+			TypeDefinitions: typedefs,
+		}
+		err := ds.WriteAuthorizationModel(ctx, store, model)
+		require.NoError(t, err)
+
+		typesys := typesystem.New(model)
+
+		listObjectsQuery := &commands.ListObjectsQuery{
```

Comment:
Note to self - we have this in a lot of places. We should consolidate to

```
func NewListObjectsQuery(...) {
    // set defaults if none are overriden (e.g. `NoopLogger`, `defaultListObjectsDeadline`, etc..)
}
```

this will cut down on boilerplate in a lot of places.